art/_config.py

```diff
@@ -2,17 +2,21 @@
 
 from __future__ import absolute_import
 
-import re
 import os
 from . import _paths
 from . import _yaml
 
 
-def save(gitlab_url, private_token):
+def save(gitlab_url, token_type, token):
     config = {
         'gitlab_url': gitlab_url,
-        'private_token': private_token,
     }
+    if token_type == 'private':
+        config['private_token'] = token
+        config['job_token'] = None
+    elif token_type == 'job':
+        config['job_token'] = token
+        config['private_token'] = None
```

Comment:
if `_config.save` never writes `private_token`, then `_config.load` can be extended as:

```python
def save():
    config = {
        'gitlab_url': gitlab_url,
        'token': token,
        'token_type': token_type,
    }
    ...
    
def load():
    config = _yaml.load(_paths.config_file)
   
    # migrate legacy private_token to token/token_type
    if 'private_token' in config:
        config['token'] = config['private_token']
        config['token_type'] = 'private'
        del config['private_token']
        
    return config
```